src/error.c

```diff
@@ -33,13 +33,49 @@
 
 
 static Obj ErrorInner;
+static Obj ERROR_OUTPUT = NULL;
 
 
 /****************************************************************************
 **
 *F * * * * * * * * * * * * * * error functions * * * * * * * * * * * * * * *
 */
 
+/****************************************************************************
+**
+*F  OpenErrorOutput()  . . . . . . . open the file or stream assigned to the
+**                                   ERROR_OUTPUT global variable defined in
+**                                   error.g, or "*errout*" otherwise
+*/
+UInt OpenErrorOutput( void )
+{
+    /* Try to print the output to stream. Use *errout* as a fallback. */
+    UInt ret = 0;
+
+    if (ERROR_OUTPUT != NULL) {
+        if (IsStringConv(ERROR_OUTPUT)) {
+            ret = OpenOutput(CONST_CSTR_STRING(ERROR_OUTPUT));
+        }
+        else {
+            ret = OpenOutputStream(ERROR_OUTPUT);
```

Comment:
We started to add more validation, and here, one could verify that ERROR_OUTPUT actually is a stream; see e.g. `PRINT_OR_APPEND_TO_FILE_OR_STREAM`.

On that note, perhaps it would be "better" to put this new function into `streams.c` or `io.c`...; but I have no strong conviction in either direction, and in any case, it's not hard to move a function to another file later on, should we feel the need for it :-).

Reply:
I wanted to check that it was actually a stream, but I couldn't quickly find the "right" way to check that.  Is there a TNUM for stream objects?  I'll look at those functions you mentioned.

I agree this function might make more sense in a `io.c`.  I put it in `error.c` because that was already loading another global variable from `error.g`.  I don't know if that entirely matters though: The connection between the C-based modules and the GAP language library modules is one that I haven't deeply explored yet and is unclear to me.

Reply:
In general, TNUMs only cover a very small, hard-wired list of types with special representations in the GAP kernel (things like integers, booleans, permutations). For things like streams, you want to use the more general `IsOutputStream`. As Max mentioned, you can see how to get this GAP level variable, and copy a reference to it into C, in PRINT_OR_APPEND_TO_FILE_OR_STREAM

Reply:
Yes, clearly I did not even test my last commit.  I'm just going to delete it.